public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx

```diff
@@ -313,10 +344,21 @@ export class PanelEditorUnconnected extends PureComponent<Props> {
   renderWithOptionsPane(styles: EditorStyles) {
     const { uiState } = this.props;
 
+    // Limit options pane width to 90% of screen.
+    const maxWidth = window.innerWidth * 0.9;
+
+    // Need to handle when width is relative. ie a percentage of the viewport
+    const width =
+      typeof uiState.rightPaneSize === 'string'
+        ? (parseFloat(uiState.rightPaneSize) / 100) * window.innerWidth
+        : uiState.rightPaneSize;
```

Comment:
With my comment above you can just assume the persisted state is stored as numbers and change this condition to check whether the number is <1 (storing ration) or >1 (storing actual width)